frontend/src/metabase/core/components/TabButton/TabButton.styled.tsx

```diff
@@ -51,7 +51,9 @@ export const TabButtonRoot = styled.div<TabButtonProps>`
   height: 1.875rem;
   padding: 0.25rem;
   color: ${props =>
-    props.isSelected && !props.disabled ? color("brand") : color("text-dark")};
+    props.isSelected && !props.disabled
+      ? "var(--mb-color-brand)"
+      : "var(--mb-color-text-primary)"};
```

Comment:
text-primary is actually text-dark in the light theme
https://github.com/metabase/metabase/blob/380a4ca520dd44517e38704f5c84dfdef1ffecca/frontend/src/metabase/css/core/colors.module.css#L33-L34

Light mode:
Before:
<img width="498" alt="image" src="https://github.com/user-attachments/assets/505387ce-6f6d-4d3d-bfc8-8d861cfd51a5">

After:
<img width="567" alt="image" src="https://github.com/user-attachments/assets/a97c09e6-ee38-4834-9a05-37afe3841b55">


In the night mode:
Before:
<img width="522" alt="image" src="https://github.com/user-attachments/assets/2a4eca0c-651b-4ab3-a484-0b64ee985e43">


After:
<img width="541" alt="image" src="https://github.com/user-attachments/assets/ca2e5743-5639-49a1-8096-378760659457">